PURL-TYPES.rst

```diff
@@ -80,20 +80,6 @@ deb
       pkg:deb/debian/dpkg@1.19.0.4?arch=amd64&distro=stretch
       pkg:deb/ubuntu/dpkg@1.19.0.4?arch=amd64
 
-docker
```

Comment:
The removal of 'docker' as a valid purl type will break compatibility for millions of users who have been using this type for the past several years.

Reply:
Ok, we can leave docker in to be backwards compatible. Would you be OK with adding a comment within the docker section suggesting to use `oci-artifact` instead of `docker`? A docker image is a type of oci artifact.

Reply:
If I may: the addition of the `docker` portion of the spec was added in 2021 by @iamwillbar (and I would hazard a guess that it was after previous conversations on how to represent container images in the purl spec). `oci-artifact` is the packaging format used by containers built by tools like docker. This change has been discussed with the OCI community here: https://hackmd.io/El8Dd2xrTlCaCG59ns5cwg#September-8-2021

Reply:
@nishakm that's not quite correct, it was added in 2017: https://github.com/package-url/purl-spec/commit/cdf28ffb1cd91424d823dcd591999fbca0f10d8f. My change was purely to the formatting of the document.

Reply:
The commit adding the "docker" examples:
```
commit 2197344e22a93d715256228b4a0e2069507410d7
Author: William Bartholomew <iamwillbar@github.com>
Date:   Tue Jun 15 13:51:43 2021 -0700

    Formatting of PURL-TYPES.rst
    
    Formatting changes, no semantic changes intended:
    * Adds headings for each type so that hyperlinks are automatically generated. Resolves #106.
    * Replaces backticks with double backticks as required by reStructuredText.
    * Adds the occasional Oxford comma for good measure.

```

Reply:
@nishakm those are from the 2017 version of the file: https://github.com/package-url/purl-spec/commit/cdf28ffb1cd91424d823dcd591999fbca0f10d8f#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168fR460. I just moved them and re-formatted them 🙂 .

Reply:
Thanks @iamwillbar! I stand corrected about the date and original author. @pombredanne, Do you remember where you got the notion of adding "docker" as the package-type?

Reply:
PS: I am OK leaving "docker" in the spec for now, provided it is understood that the canonical repository is "hub.docker.com".

Reply:
The spec does define the default repository as `hub.docker.com`, but it does allow a different registry to be specified with the `repository_url` parameter. It might be worth having a "See Also" in the `docker` section that refers to the new `oci-artifact` as a more general option.

Reply:
So this is the rub for including `docker` as a `package-type` at all. Even the tool "Docker" uses the OCI image specification as the format. There is no canonical container image repository. This ecosystem might as well be synonymous with storing `.exe` files on Sourceforge. If I can extend that analogy, the `oci-artifact` is like `http` (the transport) and `docker` is like `wget` (the command line tool).

Reply:
when I read "docker" for docker compat registries I read docker v1 registry api and docker v2 registry api (near 1/1 with oci distribution api) for docker schema format container images and also depending on release of the docker cli and hub/registry code probably also supporting oci schema formatted objects 

When I read oci for oci compatible registries images, I read more focused on just the oci v1 distribution api for storing all of the oci v1 base types (index/manifest/blob/header/config...) and not to include the similar docker v2 format.

It should be noted that docker v1 isn't widely used anymore with the focus on container images now being on docker v2 and oci v1.